// my-sql/dist/dbManager.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; /**
                                                                                                                                                                                                                                                                   * Created by pubudud on 2/25/17.
                                                                                                                                                                                                                                                                   */

var _mysql = require('mysql');

/**
 * Mysql DB Manager which creates sql connection pools and expose methods to fetch them
 */
class DbManager {

    /**
     * Initiate an active session with a default connection pool
     * @param {Object} dbConfig - mysql db configuration
     */
    init(dbConfig) {
        this.pools = {
            default: {
                default: (0, _mysql.createPool)(dbConfig)
            }
        };
        this.dbConfig = dbConfig;
    }

    /**
     * Get a connection pool given the pool name
     *
     * @param {string} [poolName] - key to identify the pool
     * @param {string} [database] - optional database name if the pool is not in the default db
     * @returns {object} - Requested connection pool
     */
    getConnectionPool(poolName, database = 'default') {
        if (!poolName) {
            return this.pools[database].default;
        }
        if (!this.pools[database] || !this.pools[database][poolName]) {
            return null;
        }
        return this.pools[database][poolName];
    }

    /**
     * Create new connection Pool
     * @param {string} name - connection pool name
     * @param {number} [size] - connection limit
     * @param {object} [overrideConfig] - optional db config if needed to override the default; possibly for a separate database
     */
    createConnectionPool(name, size, overrideConfig) {
        let dbConfig = overrideConfig || this.dbConfig;

        dbConfig = _extends({}, dbConfig, {
            connectionLimit: size || dbConfig.connectionLimit
        });

        const database = (overrideConfig || {}).database || 'default';

        if (!this.pools[database]) {
            this.pools[database] = {};
        }

        this.pools[database][name] = (0, _mysql.createPool)(dbConfig);
    }

    /**
     * End the session by terminating all available connection pools
     */
    end() {
        Object.keys(this.pools).forEach(dbKey => {
            Object.keys(this.pools[dbKey]).forEach(poolKey => {
                this.pools[dbKey][poolKey].end();
            });
        });
    }
}

exports.default = new DbManager();
