// my-sql/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.removeMiddleware = exports.setMiddleware = exports.addInternalPool = exports.endSession = exports.initSession = exports.executeTransaction = exports.executeQuery = undefined;

var _mysql = require('mysql');

Object.keys(_mysql).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _mysql[key];
    }
  });
});

var _mysqlFacade = require('./mysqlFacade');

var _mysqlFacade2 = _interopRequireDefault(_mysqlFacade);

var _dbManager = require('./dbManager');

var _dbManager2 = _interopRequireDefault(_dbManager);

var _builder = require('./query/builder');

var _builder2 = _interopRequireDefault(_builder);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Error.prototype.appendDetails = function (className, method, cause) {
  this.path = `${this.path || '#'} -> [${className}]|(${method})`;
  this.causes = `${this.causes || '#'} -> (${method})|${cause}`;
}; /**
    * Created by pubudud on 8/29/17.
    */

const executeQuery = exports.executeQuery = _mysqlFacade2.default.execute.bind(_mysqlFacade2.default);
const executeTransaction = exports.executeTransaction = _mysqlFacade2.default.executeTransaction.bind(_mysqlFacade2.default);

const initSession = exports.initSession = _dbManager2.default.init.bind(_dbManager2.default);
const endSession = exports.endSession = _dbManager2.default.end.bind(_dbManager2.default);

const addInternalPool = exports.addInternalPool = _dbManager2.default.createConnectionPool.bind(_dbManager2.default);
const setMiddleware = exports.setMiddleware = _builder2.default.use.bind(_builder2.default);
const removeMiddleware = exports.removeMiddleware = _builder2.default.removeMiddleware.bind(_builder2.default);
