// my-sql/dist/__test__/mysqlStubManager.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _sinon = require('sinon');

var _sinon2 = _interopRequireDefault(_sinon);

var _mysql = require('mysql');

var _mysql2 = _interopRequireDefault(_mysql);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Stub Manager for Device Dao
 */
/**
 * Created by pubudud on 9/4/17.
 */

class DeviceDaoStubManager {

    /**
     * Constructor
     */
    constructor() {
        this.sandbox = _sinon2.default.sandbox.create();
    }

    /**
     * Initialize all stubs of this module
     */
    initStubs() {}
    // this.restoreStubs();
    // this.initCreatePool();


    /**
     * Restore all stubs of this module
     */
    restoreStubs() {
        this.sandbox.restore();
    }

    /**
     * Init stub for getFirstDevice method
     * @returns {Object} - stub for reference
     */
    initCreatePool() {
        const createPool = this.sandbox.stub(_mysql2.default, 'createPool');

        const getConnectionStub = this.sandbox.stub();

        getConnectionStub.returns({
            execute: _sinon2.default.spy()
        });

        createPool.returns({
            getConnection: getConnectionStub,
            end: _sinon2.default.spy()
        });

        return createPool;
    }

}

exports.default = new DeviceDaoStubManager();
